src/Xml/Dom/Configurator/canonicalize.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace VeeWee\Xml\Dom\Configurator;
+
+use DOMDocument;
+use function Psl\Fun\pipe;
+use function VeeWee\Xml\Dom\Loader\xml_string_loader;
+use function VeeWee\Xml\ErrorHandling\disallow_libxml_false_returns;
+
+/**
+ * @return callable(DOMDocument): DOMDocument
+ */
+function canonicalize(): callable
```

Comment:
To investigate : namespace optimizations:

See https://genericparallel.com/blog/2018/02/optimize-namespaces-to-reduce-xml-size/

This won't be an easy fix!